src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -884,6 +888,67 @@ class BookingService(
       )
     }
 
+    if (booking.service == ServiceName.approvedPremises.value && booking.application != null && user != null) {
+      val dateTime = OffsetDateTime.now()
+
+      val domainEventId = UUID.randomUUID()
+
+      val offenderDetails = when (val offenderDetailsResult = offenderService.getOffenderByCrn(booking.crn, user.deliusUsername, user.hasQualification(UserQualification.LAO))) {
```

Comment:
Given this is going to an internal service, do we need to worry about if the user has the qualification here?

Reply:
Good point, have updated this & added another commit to address for the existing events